### PR TITLE
Case 22249: migrate to Qt 5.12.3

### PR DESCRIPTION
**Pull request by NissimHadar**
https://highfidelity.fogbugz.com/f/cases/22249/Migrate-to-Qt-5-12-3

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18784/HighFidelity-Beta-PR15408-25f4016.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18784/HighFidelity-Beta-Interface-PR15408-25f4016.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18784/HighFidelity-Beta-PR15408-25f4016.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18784/HighFidelity-Quest-Beta-PR15408-25f4016.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18789/HighFidelity-Beta-PR15408-2aabf02.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18789/HighFidelity-Beta-Interface-PR15408-2aabf02.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18789/HighFidelity-Beta-PR15408-2aabf02.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18789/HighFidelity-Beta-Interface-PR15408-2aabf02.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18789/HighFidelity-Beta-PR15408-2aabf02.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18789/HighFidelity-Quest-Beta-PR15408-2aabf02.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18869/HighFidelity-Beta-PR15408-79fd591.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18869/HighFidelity-Beta-Interface-PR15408-79fd591.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18869/HighFidelity-Beta-PR15408-79fd591.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18869/HighFidelity-Beta-Interface-PR15408-79fd591.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18869/HighFidelity-Beta-PR15408-79fd591.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18869/HighFidelity-Quest-Beta-PR15408-79fd591.apk)

File: BUILD.md [hunk 1]
```diff
@@ -7,8 +7,8 @@
 
 ### Dependencies
 
-- [cmake](https://cmake.org/download/):  3.9
-- [Qt](https://www.qt.io/download-open-source):   5.10.1
+- [cmake](https://cmake.org/download/):  3.13
+- [Qt](https://www.qt.io/download-open-source):   5.12.2
```
Comment (AndrewMeadows) [inline BUILD.md]: Should be `5.12.3`?

File: BUILD_LINUX.md [hunk 2]
```diff
@@ -80,7 +82,7 @@ cd hifi/build
 
 Prepare makefiles:
 ```bash
-cmake -DQT_CMAKE_PREFIX_PATH=/usr/local/Qt5.10.1/5.10.1/gcc_64/lib/cmake ..
+cmake -DQT_CMAKE_PREFIX_PATH=~/Qt/5.12.2/gcc_64/lib/cmake ..
```
Comment (AndrewMeadows) [inline BUILD_LINUX.md]: Should be `5.12.3`?

File: BUILD_WIN.md [hunk 3]
```diff
@@ -85,4 +90,4 @@ Remove `CMakeCache.txt` found in the `%HIFI_DIR%\build` directory.  Verify that
 
 #### Qt is throwing an error
 
-Make sure you have the correct version (5.10.1) installed and `QT_CMAKE_PREFIX_PATH` environment variable is set correctly.
+Make sure you have the correct version (5.12.2) installed and `QT_CMAKE_PREFIX_PATH` environment variable is set correctly.
```
Comment (AndrewMeadows) [inline BUILD_WIN.md]: Should be `5.12.3`?

File: qfloat16.h [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+Q_REQUIRED_RESULT inline bool qFuzzyIsNull(qfloat16 f) Q_DECL_NOTHROW
+{
+    return qAbs(static_cast<float>(f)) <= 0.001f;
+}
+
+QT_END_NAMESPACE
+
+Q_DECLARE_METATYPE(qfloat16)
+
+#endif // QFLOAT16_H
```
Comment (AndrewMeadows) [inline qfloat16.h]: What is the purpose of adding this file?  Why in the base directory?

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18975/HighFidelity-Beta-PR15408-16e22a5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/18975/HighFidelity-Beta-Interface-PR15408-16e22a5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18975/HighFidelity-Beta-PR15408-16e22a5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/18975/HighFidelity-Beta-Interface-PR15408-16e22a5.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18975/HighFidelity-Beta-PR15408-16e22a5.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/18975/HighFidelity-Quest-Beta-PR15408-16e22a5.apk)

File: qfloat16.h (hunk 4 above)
Comment (AndrewMeadows) [reply qfloat16.h]: Ah... I now understand.  I tried to build on windows with VS2017, saw the failed build for myself, diffed qfloat16.h with what is in the Qt install, read the bug report, and finally spotted the cmake config change that defines QT_NO_FLOAT16_OPERATORS.

**Comment by NissimHadar** [comment]
VS 2019 solves that bug, so it's good to remember the patch can be removed
in thr future.

**Comment by Triplelexx** [comment]
There's still a number of unchanged references to v5.10.1 in the build documents.

**Comment by AndrewMeadows** [comment]
Closing this PR.  I'll create a new one with an updated branch.

**Pull request closed without merging**